plugin/trino-tpch/src/main/java/io/trino/plugin/tpch/TpchConnectorFactory.java

```diff
@@ -45,11 +45,11 @@
     public static final String TPCH_TABLE_SCAN_REDIRECTION_CATALOG = "tpch.table-scan-redirection-catalog";
     public static final String TPCH_TABLE_SCAN_REDIRECTION_SCHEMA = "tpch.table-scan-redirection-schema";
     public static final String TPCH_SPLITS_PER_NODE = "tpch.splits-per-node";
+    public static final String TPCH_PARTITIONING_ENABLED = "tpch.partitioning-enabled";
```

Comment:
The AddDynamicFilterSource rule relies on presence of remote exchange to add the DynamicFilterSourceNode, whereas in pipeline execution the dynamic filter stays as part of join node.
The presence of pre-partitioning makes the remote exchange unnecessary, so AddDynamicFilterSource isn't able to add DynamicFilterSourceNode to the plan. Currently I get rid of the dynamic filter in the join even if we're not able to convert it into DynamicFilterSourceNode for simplicity. 
I think we could improve it to allow keeping the dynamic filter in the join for such cases, but it would easier to tackle that as a follow up.

Reply:
I see. I think for collocated joins it should still work? Right? Since there's no need to involve coordinator in the process (e.g.: local dynamic filter)

Reply:
The co-located joins case already works in pipeline mode, it can be made to work in fault tolerant mode as well. The difficulty is that right now I remove DFs from the JoinNode even if the rewrite of DFs to DynamicFilterSourceNode couldn't be done because this handles all the unsupported/missed cases cleanly. To make DF work for co-located joins I think I would have to leave DF in the JoinNode when the rewrite to DynamicFilterSourceNode doesn't succeed and distinguish it somehow from the other unsupported/missed cases. 
I'm thinking we can tackle this later on after landing this PR.